pkg/controller/dual-pods/inference-server.go

```diff
@@ -475,3 +577,15 @@ func SliceMap[Domain, Range any](slice []Domain, mapFn func(Domain) (Range, erro
 	}
 	return mapped, errors
 }
+
+func IterFilter[Elt any](it iter.Seq[Elt], include func(Elt) bool) iter.Seq[Elt] {
+	return func(yield func(Elt) bool) {
+		for elt := range it {
+			if include(elt) {
+				if !yield(elt) {
+					return
+				}
+			}
+		}
+	}
+}
```

Comment:
I find this nested `if` a little awkward. Can we combined the `if` statements?:
```suggestion
func IterFilter[Elt any](it iter.Seq[Elt], include func(Elt) bool) iter.Seq[Elt] {
	return func(yield func(Elt) bool) {
		for elt := range it {
			if include(elt) && !yield(elt) {
				return
			}
		}
	}
}
```